tests/series/test_size_bytes.py

```diff
@@ -13,233 +13,167 @@
-)
 @pytest.mark.parametrize("size", [0, 1, 2, 8, 9, 16])
-def test_series_binary_size_bytes(size) -> None:
+@pytest.mark.parametrize("with_nulls", [True, False])
+def test_series_binary_size_bytes(size, with_nulls) -> None:
     pydata = [str(i).encode("utf-8") for i in range(size)]
-    data = pa.array(pydata, pa.large_binary())
-    s = Series.from_arrow(data)
 
-    assert s.datatype() == DataType.binary()
-    # TODO(Clark): This is required due to an off-by-one error in pyarrow's calculation of of the offset array length.
-    # We should fix this upstream and/or refactor these tests to not rely on pyarrow as the source-of-truth.
-    assert s.size_bytes() == data.nbytes + 8
-
-    ## with nulls
-    if size > 0:
-        pydata = pydata[:-1] + [None]
-    data = pa.array(pydata, pa.large_binary())
+    if with_nulls and size > 0:
+        data = pa.array(pydata[:-1] + [None], pa.large_binary())
+    else:
+        data = pa.array(pydata, pa.large_binary())
+
     s = Series.from_arrow(data)
 
     assert s.datatype() == DataType.binary()
-    assert s.size_bytes() == data.nbytes + 8
+    assert s.size_bytes() == get_total_buffer_size(data)
 
 
-@pytest.mark.skipif(
-    not PYARROW_GE_7_0_0,
-    reason="Array.nbytes behavior changed in versions >= 7.0.0. Old behavior is incompatible with our tests and is renamed to Array.get_total_buffer_size()",
-)
 @pytest.mark.parametrize("dtype, size", itertools.product(ARROW_INT_TYPES + ARROW_FLOAT_TYPES, [0, 1, 2, 8, 9, 16]))
-def test_series_list_size_bytes(dtype, size) -> None:
+@pytest.mark.parametrize("with_nulls", [True, False])
+def test_series_list_size_bytes(dtype, size, with_nulls) -> None:
     list_dtype = pa.list_(dtype)
     pydata = [[2 * i] if i % 2 == 0 else [2 * i, 2 * i + 1] for i in range(size)]
-    data = pa.array(pydata, list_dtype)
+
+    if with_nulls and size > 0:
+        data = pa.array(pydata[:-1] + [None], list_dtype)
+    else:
+        data = pa.array(pydata, list_dtype)
 
     s = Series.from_arrow(data)
 
-    assert s.datatype() == DataType.from_arrow_type(list_dtype)
-    # TODO(Clark): Investigate this discrepancy.
-    # TODO(Clark): Investigate this discrepancy between Arrow2 and pyarrow.
-    # We should fix this upstream and/or refactor these tests to not rely on pyarrow as the source-of-truth.
-    assert s.size_bytes() == data.nbytes + (size * 8) // 2
+    # Offset array is increased in bit width from 32 to 64 when converting to large_list
+    conversion_to_large_list_bytes = len(data) * 4
 
-    ## with nulls
-    if size > 0:
-        pydata = pydata[:-1] + [None]
-    data = pa.array(pydata, list_dtype)
+    size_bytes = s.size_bytes()
 
-    s = Series.from_arrow(data)
+    # TODO(jay): There is an off-by-1 error in the arrow2 estimated_bytes_size API for List and LargeList:
+    # https://github.com/jorgecarleitao/arrow2/blob/64d8ec203f991468032025a13a4f971f1f2cfc14/src/compute/aggregate/memory.rs#LL76C1-L76C1
```

Comment:
Nice catch! Those should totally be `offsets.len()` rather than `offsets.len_proxy()`.

Reply:
Yep once we do the arrow2 fork we should upstream a fix!